src-tauri/src/main.rs

```diff
@@ -72,80 +78,91 @@ async fn control_grabbing(
     frontend_bool: bool,
 ) -> Result<(), ()> {
     state.set_is_listening(frontend_bool);
-    // set_macro_listening(state, frontend_bool);
+    Ok(())
+}
+
+/// Enables or disables the automatic startup of Wootomation at system start.
+fn init_autostart(app_name: &str, set_autolaunch: bool) -> Result<(), Error> {
+    let current_exe = std::env::current_exe().context("current EXE not found")?;
+
+    let auto_start = auto_launch::AutoLaunchBuilder::new()
+        .set_app_name(app_name)
+        .set_app_path(current_exe.as_path().to_str().unwrap())
+        .set_use_launch_agent(true)
+        .build()
+        .context("App name is empty, or unsupported OS is used.")?;
+
+    match set_autolaunch {
+        true => auto_start.enable()?,
+        false => {
+            if let Err(e) = auto_start.disable() {
+                match e {
+                    auto_launch::Error::Io(err) => match err.kind() {
+                        std::io::ErrorKind::NotFound => {
+                            trace!("Autostart is already removed, finished checking.");
+                        }
+                        _ => return Err(err.into()),
+                    },
+                    _ => return Err(e.into()),
+                }
+            }
+        }
+    }
     Ok(())
 }
 
 #[tokio::main(flavor = "multi_thread", worker_threads = 10)]
 /// Spawn the backend thread.
 /// Note: this doesn't work on macOS since we cannot give the thread the proper permissions
 /// (will crash on key grab/listen)
-async fn main() {
+async fn main() -> Result<(), Error> {
+    // The optionENV only takes a string literal and can't be forced to take a const of any type.
     let log_level: log::LevelFilter = option_env!("MACRO_LOG_LEVEL")
```

Comment:
You have a variable for this now env variable now, would be good if it can be used here as well. Also, option_env! is a compile time retrieval of the env variable, not runtime, which I don't think is exactly what you want lol

Reply:
Tweaked: [7786af2](https://github.com/WootingKb/wooting-macros/pull/194/commits/7786af2c06c7aeb7b924462c7514821fa80381ed)